src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -2555,22 +2556,71 @@ def _extract_token_timestamps(self, generate_outputs, alignment_heads, time_prec
         # of shape (batch size, num selected, output length, input length).
         weights = torch.stack([cross_attentions[l][:, h] for l, h in alignment_heads])
         weights = weights.permute([1, 0, 2, 3])
+
+        if "beam_indices" in generate_outputs:
+            # If beam search has been used, the output sequences can have been generated for more timesteps than their sequence_lengths
+            # Since the beam search strategy choose the most probable sequences at the end of the search
+            # In that case, we make sure that the weights have the right output_length
+            weight_length = (generate_outputs.beam_indices != -1).sum(-1).max()
+            weights = weights[:, :, :weight_length]
+
+            # If beam indice is still -1, it means that the associated token id is EOS
```

Comment:
I'm not super au fait with beam generation but the comment and code here aren't clear to me:
* What do we mean 'still -1'. If we've generated EOS won't it remain -1? 
* 'associated token id' - associated with what? 
* What does setting to 0 here mean?